lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,754 @@
+mock_register_module("finetuningscheduler")
+# set notebook-level variables
+AVAIL_GPUS = torch.cuda.device_count()
+TASK_NUM_LABELS = {"boolq": 2, "rte": 2}
+DEFAULT_TASK = "rte"
+
+transformers_logging.set_verbosity_error()
+# ignore warnings related tokenizers_parallelism/DataLoader parallelism tradeoff and
+#  expected logging behavior
+for warnf in [".*does not have many workers*", ".*The number of training samples.*"]:
+    warnings.filterwarnings("ignore", warnf)
+
+
+# %%
+class RteBoolqDataModule(pl.LightningDataModule):
+    """A ``LightningDataModule`` for using either the RTE or BoolQ SuperGLUE Hugging Face datasets."""
+
+    TASK_TEXT_FIELD_MAP = {"rte": ("premise", "hypothesis"), "boolq": ("question", "passage")}
+    LOADER_COLUMNS = (
+        "datasets_idx",
+        "input_ids",
+        "token_type_ids",
+        "attention_mask",
+        "start_positions",
+        "end_positions",
+        "labels",
+    )
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        task_name: str = DEFAULT_TASK,
+        max_seq_length: int = 128,
+        train_batch_size: int = 16,
+        eval_batch_size: int = 16,
+        tokenizers_parallelism: bool = True,
+        **dataloader_kwargs: Any,
+    ):
+        super().__init__()
+        self.model_name_or_path = model_name_or_path
+        self.task_name = task_name if task_name in TASK_NUM_LABELS.keys() else DEFAULT_TASK
+        self.max_seq_length = max_seq_length
+        self.train_batch_size = train_batch_size
+        self.eval_batch_size = eval_batch_size
+        self.tokenizers_parallelism = tokenizers_parallelism
+        self.dataloader_kwargs = {
+            "num_workers": dataloader_kwargs.get("num_workers", 0),
+            "pin_memory": dataloader_kwargs.get("pin_memory", False),
+        }
+        self.text_fields = self.TASK_TEXT_FIELD_MAP[self.task_name]
+        self.num_labels = TASK_NUM_LABELS[self.task_name]
+        os.environ["TOKENIZERS_PARALLELISM"] = "true" if self.tokenizers_parallelism else "false"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True, local_files_only=False)
+
+    def setup(self, stage):
+        self.dataset = datasets.load_dataset("super_glue", self.task_name)
+        for split in self.dataset.keys():
+            self.dataset[split] = self.dataset[split].map(
+                self._convert_to_features, batched=True, remove_columns=["label"]
+            )
+            self.columns = [c for c in self.dataset[split].column_names if c in self.LOADER_COLUMNS]
+            self.dataset[split].set_format(type="torch", columns=self.columns)
+
+        self.eval_splits = [x for x in self.dataset.keys() if "validation" in x]
+
+    def prepare_data(self):
+        # N.B. PL calls prepare_data from a single process (rank 0) so do not use it to assign
+        # state (e.g. self.x=y)
+        datasets.load_dataset("super_glue", self.task_name)
```

Comment:
```suggestion
    def prepare_data(self):
        # N.B. PL calls prepare_data from a single process (rank 0) so do not use it to assign
        # state (e.g. self.x=y)
        datasets.load_dataset("super_glue", self.task_name)
 
    def setup(self, stage):
        self.dataset = datasets.load_dataset("super_glue", self.task_name)
        for split in self.dataset.keys():
            self.dataset[split] = self.dataset[split].map(
                self._convert_to_features, batched=True, remove_columns=["label"]
            )
            self.columns = [c for c in self.dataset[split].column_names if c in self.LOADER_COLUMNS]
            self.dataset[split].set_format(type="torch", columns=self.columns)

        self.eval_splits = [x for x in self.dataset.keys() if "validation" in x]
```

Reply:
Just putting prepare_data first as it is call first :)

Reply:
nice, thanks!